client/packages/inventory/src/Stocktake/DetailView/modal/StocktakeLineEdit/StocktakeLineEdit.tsx

```diff
@@ -64,48 +65,17 @@ export const StocktakeLineEdit: FC<StocktakeLineEditProps> = ({
   };
 
   const onOk = async () => {
-    const result = await save(draftLines);
-
-    result.batchStocktake?.insertStocktakeLines?.map(response => {
-      if (
-        response.response?.__typename === 'InsertStocktakeLineError' &&
-        response.response?.error?.__typename === 'StockLineReducedBelowZero'
-      ) {
-        return error(t('error.reduced-below-zero'))();
-      }
-    });
-    result.batchStocktake.insertStocktakeLines?.map(response => {
-      if (
-        response.response?.__typename === 'InsertStocktakeLineError' &&
-        response.response?.error?.__typename === 'AdjustmentReasonNotProvided'
-      ) {
-        return error(t('error.provide-reason'))();
-      }
-    });
-
-    result.batchStocktake?.updateStocktakeLines?.map(response => {
-      if (
-        response?.response?.__typename === 'UpdateStocktakeLineError' &&
-        response?.response?.error?.__typename === 'StockLineReducedBelowZero'
-      ) {
-        return error(t('error.reduced-below-zero'))();
-      }
-    });
-    result.batchStocktake.updateStocktakeLines?.map(response => {
-      if (
-        response?.response?.__typename === 'UpdateStocktakeLineError' &&
-        response?.response?.error?.__typename === 'AdjustmentReasonNotProvided'
-      ) {
-        return error(t('error.provide-reason'))();
-      }
-    });
+    let { errorMessages } = await save();
+    if (errorMessages)
+      return errorMessages.forEach(errorMessage => error(errorMessage)());
```

Comment:
It looks like previously we might have still. navigated to next item even on error